internal/handlers/common/filter.go

```diff
@@ -348,9 +348,23 @@ func filterFieldExpr(doc *types.Document, filterKey string, expr *types.Document
 
 		case "$gt":
 			// {field: {$gt: exprValue}}
-			if _, ok := exprValue.(types.Regex); ok {
+			switch exprValue.(type) {
+			case *types.Document, *types.Array:
+			case types.Regex:
 				msg := fmt.Sprintf(`Can't have RegEx as arg to predicate over field '%s'.`, filterKey)
 				return false, NewCommandErrorMsgWithArgument(ErrBadValue, msg, exprKey)
+			default:
+				if arrValue, ok := fieldValue.(*types.Array); ok {
+					// Filter array by only keeping the same type as exprValue.
```

Comment:
As discussed in the call, we need the "why" for this comment. 
For example, something like "only values of the same type can be compared, so we filter the array" (but maybe there is a better wording for it"

Reply:
I added example as well :pray: